logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/PersistMessageUseCase.kt

```diff
@@ -12,22 +16,34 @@ interface PersistMessageUseCase {
     suspend operator fun invoke(message: Message.Standalone): Either<CoreFailure, Unit>
 }
 
+@OptIn(DelicateKaliumApi::class)
 internal class PersistMessageUseCaseImpl(
     private val messageRepository: MessageRepository,
-    private val selfUser: UserId
+    private val userRepository: UserRepository
 ) : PersistMessageUseCase {
     override suspend operator fun invoke(message: Message.Standalone): Either<CoreFailure, Unit> {
-        val isMyMessage = message.senderUserId == selfUser
-        @OptIn(DelicateKaliumApi::class)
+        val (updateConversationNotificationsDate, isMyMessage) = userRepository.getSelfUser()?.let {
+            message.shouldUpdateConversationNotificationDate(it) to message.isMyMessage(it.id)
+        } ?: (false to false)
         return messageRepository
             .persistMessage(
                 message = message,
                 updateConversationReadDate = isMyMessage,
                 updateConversationModifiedDate = message.content.shouldUpdateConversationOrder(),
-                updateConversationNotificationsDate = isMyMessage
+                updateConversationNotificationsDate
             )
     }
 
+    private fun Message.shouldUpdateConversationNotificationDate(selfUser: SelfUser) =
+        when (selfUser.availabilityStatus) {
+            UserAvailabilityStatus.AWAY -> this.isMyMessage(selfUser.id)
+            UserAvailabilityStatus.BUSY -> true // todo: OR conversationMutedStatus == MutedConversationStatus.OnlyMentionsAndRepliesAllowed
```

Comment:
will you take care of the TODO in this PR ?

Reply:
@Garzas is working on it, maybe we can have it in this PR! But is not originally for this PR [it wasn't supported before]